templates/common/_base/files/ofport-request.yaml

```diff
@@ -0,0 +1,112 @@
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/pre-up.d/10-ofport-request.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    # Set interface ofport_request to guarantee stable ofport numbers. This is important for flow matches.
+    # Otherwise, another ofport number is assigned to the interface on every restart of NetworkManager.
+    # This script will build a map INTERFACE_NAME->ofport_request and will save it to file CONFIGURATION_FILE.
+    # When an interface is brought up, this will reuse the value from the map if such a value exists.
+    # Otherwise, this will try to use the current ofport value. If the ofport value is already reserved, then
+    # this use the lowest available numerical value, instead.
+    set -eux -o pipefail
+    if [[ "{{ .NetworkType }}" != "OVNKubernetes" ]]; then
+        exit 0
+    fi
+    
+    INTERFACE_NAME=$1
+    OPERATION=$2
+
+    # Only execute this on pre-up
+    if [ "${OPERATION}" != "pre-up" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's NM uuid
+    INTERFACE_CONNECTION_UUID=$(nmcli -t -f  device,type,uuid conn | awk -F ':' '{if($1=="'${INTERFACE_NAME}'" && $2!~/^ovs*/) print $NF}')
+    if [ "$INTERFACE_CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's slave-type. If this is not an ovs-port, then exit
+    INTERFACE_OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$INTERFACE_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$INTERFACE_OVS_SLAVE_TYPE" != "ovs-port" ]; then
+        exit 0
+    fi
+    
+    # This is not necessarily a UUID (can be a name in case of bonds) but this should be unique
+    PORT=$(nmcli -t -f connection.master conn show "$INTERFACE_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's NM uuid
+    PORT_CONNECTION_UUID=$(nmcli -t -f  device,type,uuid conn | awk -F ':' '{if( ($1=="'${PORT}'" || $3=="'${PORT}'") && $2~/^ovs*/) print $NF}')
+    if [ "$PORT_CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the port's slave-type. If this is not an ovs-bridge, then exit
+    PORT_OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT_OVS_SLAVE_TYPE" != "ovs-bridge" ]; then
+        exit 0
+    fi
+    
+    # Get the port's master = bridge name
+    BRIDGE_NAME=$(nmcli -t -f connection.master conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$BRIDGE_NAME" == "" ]; then
+        exit 0
+    fi
+    
+    CONFIGURATION_FILE=/etc/NetworkManager/ofport_requests.${BRIDGE_NAME}.conf
```

Comment:
Hmm probably not the best place for this file. If we do not care to stabilize ofports across reboots, and I guess it would be nice if we didn't, what about `/run/configure-ovs-ofport-requests.${BRIDGE_NAME}`